### PR TITLE
Run clusterWriteLoop on the master topic level.

### PR DESCRIPTION
**Pull request by aforge**
Remove manual scheduling.

This is a draft version of it. Please, take a look and let me know what you think.

### CODE REVIEW

File: server/cluster.go [hunk 1]
```diff
@@ -1117,24 +1117,36 @@ func (c *Cluster) gcProxySessionsForNode(node string) {
 
 // clusterWriteLoop implements write loop for multiplexing (proxy) session at a node which hosts master topic.
```
Comment (or-else) [inline server/cluster.go:1118]: The comment looks outdated.

File: server/topic.go [hunk 2]
```diff
@@ -46,6 +47,19 @@ type Topic struct {
...
+	proxiedSessions []*Session
+	// Proxied sessions' channels for the use in the topic's clusterWriteLoop:
+	// i-th session's channels (proxiedSessions[i]) are found at:
+	// proxiedChannels[i * 3 + 1] - send
+	// proxiedChannels[i * 3 + 2] - stop
+	// proxiedChannels[i * 3 + 3] - detach
+	//
+	// proxiedChannels[0] is a special-purpose channel necessary for interrupting
+	// clusterWriteLoop when sessions are added or removed.
+	proxiedChannels []reflect.SelectCase
```
Comment (or-else) [inline server/topic.go:62]: I don't understand why Reflect. Why not just use a struct? It makes things hard to read without adding clear value.

Comment (or-else) [reply server/topic.go:62]: Ah, I understand now. You want to use a single select to check all channels in one place. Interesting.

File: server/topic.go [hunk 3]
```diff
@@ -3320,6 +3334,57 @@ func (t *Topic) subsCount() int {
 	return len(t.perUser)
 }
 
+// Adds a new multiplex proxied session to the topic's clusterWriteLoop.
+func (t *Topic) addProxiedSession(s *Session) {
```
Comment (or-else) [inline server/topic.go:3338]: reflect.Select can select from up to 2^16 channels. Is it possible to have more than 2^14 sessions?

Comment (or-else) [reply server/topic.go:3338]: Or maybe these are multiplexing sessions, i.e. one per cluster member?

File: server/cluster.go (hunk 1 above)
Comment (aforge) [reply server/cluster.go]: Done.

File: server/topic.go (hunk 2 above)
Comment (aforge) [reply server/topic.go:62]: Correct.

File: server/topic.go (hunk 3 above)
Comment (aforge) [reply server/topic.go:3338]: The max number of multiplexing sessions we have per topic is `#number of nodes - 1`.
In practice, we should never go all the way up to `2^16`. Even 10-20 nodes is already a rather large footprint.

**Review by or-else — APPROVED** [review]

**Pull request merged**